File: pkg/middleware/accept_language.go
// Copyright (c) 2026 LingByte. All rights reserved.
// SPDX-License-Identifier: AGPL-3.0

package middleware

import (
	"strings"

	"github.com/LingByte/LingVoice/i18n"
	"github.com/gin-gonic/gin"
)

// AcceptLanguage parses Accept-Language and stores a normalized tag on the Gin context
// for lingi18n.GetLangFromContext (see i18n.GinContextKeyLanguage).
func AcceptLanguage() gin.HandlerFunc {
	return func(c *gin.Context) {
		h := strings.TrimSpace(c.GetHeader("Accept-Language"))
		lang := i18n.ParseAcceptLanguage(h)
		if lang != "" {
			c.Set(i18n.GinContextKeyLanguage, lang)
		}
		c.Next()
	}
}

File: web/src/locale/acceptLanguage.ts
// Copyright (c) 2026 LingByte. All rights reserved.
// SPDX-License-Identifier: AGPL-3.0

import type { AppLocale } from '@/locale/storage'

/** Value for `Accept-Language`; backend `i18n.ParseAcceptLanguage` reads the first tag. */
export function acceptLanguageHeader(locale: AppLocale): string {
  switch (locale) {
    case 'zh-CN':
      return 'zh-CN,zh;q=0.9,en;q=0.8,ja;q=0.5'
    case 'ja':
      return 'ja,en;q=0.85,zh-CN;q=0.7'
    default:
      return 'en,zh-CN;q=0.85,ja;q=0.6'
  }
}
